### PR TITLE
Add company pages

### PR DESCRIPTION
**Pull request by oleast**
For now just created some similar styling to the `hobbygroups` pages to make company cards for a list

Depends on https://github.com/dotkom/onlineweb4/pull/2473 for data

Resvoles #194

### CODE REVIEW

**Comment by oleast** [comment]
Some things are still missing, but I would still appreciate a review already :smile: 

Still missing:
- [] Finishing touches on styling for company detail page
- [] Some actual search functionality for the company list page

There should also be some way to link to the company list page, but that is part of a larger problem of linking to all our services #50

File: src/companies/components/CompanyDetail/CompanyCareerOpportunities/CareerOpportunity.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export const CareerOppotunity: FC<IProps> = ({ oppotunity }) => {
+  const deadline = DateTime.fromISO(oppotunity.deadline).toLocaleString();
+  return (
+    <Link {...getCareerOpportinityUrl(oppotunity.id)}>
+      <a className={style.container}>
+        <p className={style.name}>{oppotunity.employment.name}</p>
+        <p className={style.count}>{oppotunity.title}</p>
+        <p>{oppotunity.location.map((location) => location.name)}</p>
+        <p className={style.count}>{deadline}</p>
```
Comment (henrikhorluck) [inline src/companies/components/CompanyDetail/CompanyCareerOpportunities/CareerOpportunity.tsx:22]:
```suggestion
        <p className={style.count}>{deadline || "Ikke spesifisert"}</p>
```

File: src/companies/components/CompanyList/NextPageObserver.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  return state.companies.search.loading === 'pending';
+};
+
+const selectTotalCount = () => (state: State) => {
+  return state.companies.search.count;
+};
+
+const selectCurrentCount = () => (state: State) => {
+  return state.companies.search.ids.length;
+};
```
Comment (FluidSense) [inline src/companies/components/CompanyList/NextPageObserver.tsx:47]: This observer, is this what you talked about which would allow for smoother population and rerending?

Comment (oleast) [reply src/companies/components/CompanyList/NextPageObserver.tsx:47]: It's been a while since I wrote this, I can't recall what you are referring to?

**Comment by oleast** [comment]
> 1. There's an awful lot of API-calls for searches. Could we implement some kind of cache so that only "new" queries for a session is called? So that "s", "se", "s" would be reduced to 2 calls instead of 1, since "s" has already been queried and should be in the story? Perhaps with some timestamp to allow for requerying if you have been on the page for a long time?

That could be possible to solve in some way. Caching is a huge thing in and of itself, and we should look into default caching for `fetch` requests. It can be done in multiple ways, but I think there is built in support for it in ServiceWorkers?

> 2. Could subheaders be more prominent, perhaps bolded?
>    I found it a bit difficult to notice the sections "Arrangementer" and "Aktive jobbannonser".

I agree on that one. The page here is just rendered markdown just as all other pages that use it. The font becoming this small is because the inclusion of Design System. Could potentially be solved by using the markdown renderer from that project?

File: src/companies/components/CompanyList/NextPageObserver.tsx (hunk 2 above)
Comment (FluidSense) [reply src/companies/components/CompanyList/NextPageObserver.tsx:47]: Let me rephrase: What exactly is a NextPageObserver? Is it a standard Next-thing?

Comment (oleast) [reply src/companies/components/CompanyList/NextPageObserver.tsx:47]: Ah, no. It's a thing I wrote to listen for when the next "_page_" of an infinite scrolling list should be requested
